src/Expander/Primitives.hs

```diff
@@ -406,7 +406,7 @@ listSyntax :: ExprPrim
 listSyntax t dest stx = do
   unify dest t tSyntax
   Stx _ _ (_, list, source) <- mustHaveEntries stx
-  Stx _ _ listItems <- mustHaveEntries list
+  listItems <- mustHaveShape list
```

Comment:
This was the only place where we were using `mustHaveEntries` to assert that a Syntax is a List. I have dropped support for this feature, add is it already supported by `mustHaveShape`.

`mustHaveEnteies` now only supports checking that it is a List with a required number of elements, but I have generalized `mustHaveEntries` so that this required number can be "Either N or M" and not just a single number N.

Reply:
This seems like a nice change.